spec/rails_helper.rb

```diff
@@ -65,6 +66,7 @@
   config.include Devise::Test::ControllerHelpers, type: :controller
   config.include Warden::Test::Helpers
   config.include Devise::Test::IntegrationHelpers, type: :request
+  config.extend ControllerMacros, :type => :controller
```

Comment:
[as per the Devise documentation](https://github.com/heartcombo/devise/wiki/How-To:-Test-controllers-with-Rails-(and-RSpec)#controller-specs).

Reply:
Minor, but i would use consistent ruby formatting here

```rb
:type => :controller
vs
type: :controller
```

Reply:
Amended